index.js

```diff
@@ -2,20 +2,25 @@ var domToReact = require('./lib/dom-to-react');
 var htmlToDOM = require('html-dom-parser');
 
 // decode HTML entities by default for `htmlparser2`
-var domParserOptions = { decodeEntities: true, lowerCaseAttributeNames: false };
+var defaultDomParserOptions = { decodeEntities: true, lowerCaseAttributeNames: false };
 
 /**
  * Convert HTML string to React elements.
  *
  * @param  {String}   html              - The HTML string.
  * @param  {Object}   [options]         - The additional options.
  * @param  {Function} [options.replace] - The replace method.
+ * @param  {Object}   [options.domParserOptions] - Additional domParserOptions options.
  * @return {ReactElement|Array}
  */
 function HTMLReactParser(html, options) {
   if (typeof html !== 'string') {
     throw new TypeError('First argument must be a string');
   }
+  
+  // adding new options to domParserOptions
+  var domParserOptions = Object.assign(defaultDomParserOptions, (options && options.domParserOptions) || {});
```

Comment:
**fix:** the logic here mutates the current default options.

Also, [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Browser_compatibility) isn't supported in IE11 and older browsers.

You may want to update the logic to:

```js
var domParserOptions = defaultDomParserOptions;
if (options.domParserOptions instanceof Object) {
  domParserOptions = options.domParserOptions;
  for (var key in defaultDomParserOptions) {
    if (domParserOptions[key] === undefined) {
      domParserOptions[key] = defaultDomParserOptions[key];
    }
  }
}
```

Reply:
I added `options` check, final:
```
  var domParserOptions = defaultDomParserOptions;
  if (options && options.domParserOptions instanceof Object) {
    domParserOptions = options.domParserOptions;
    for (var key in defaultDomParserOptions) {
      if (domParserOptions[key] === undefined) {
        domParserOptions[key] = defaultDomParserOptions[key];
      }
    }
  }
```